app/model/commands/PublishAtomCommand.scala

```diff
@@ -92,20 +89,68 @@ case class PublishAtomCommand(id: String, override val stores: DataStores, youTu
     }
   }
 
-  private def updateYouTube(atom: MediaAtom, asset: Asset): Unit = {
+  private def updateYouTube(previewAtom: MediaAtom, asset: Asset): MediaAtom = {
+    updateYoutubeMetadata(previewAtom, asset)
+    updateYoutubeThumbnail(previewAtom, asset)
+    createOrUpdateYoutubeClaim(previewAtom, asset)
+  }
+
+  private def createOrUpdateYoutubeClaim(previewAtom: MediaAtom, asset: Asset): MediaAtom = {
+    if (previewAtom.blockAds.isEmpty) {
```

Comment:
When would the blockAds field not be set? If its only for unpublished atoms, don't we check for that underneath?